src/geom/face_tri3.C

```diff
@@ -203,4 +203,50 @@ std::pair<Real, Real> Tri3::min_and_max_angle() const
                         std::max(theta0, std::max(theta1,theta2)));
 }
 
+bool Tri3::contains_point (const Point & p, Real tol) const
+{
+  // move test point relative to node 0
+  const Point p0 ( p - this->point(0) );
+
+  // define two in plane vectors
+  Point v1 ( this->point(1) - this->point(0) );
+  Point v2 ( this->point(2) - this->point(0) );
+
+#if LIBMESH_DIM == 3
+  // define out of plane vector
+  Point oop ( v1.cross(v2) );
+
+  // project moved test point onto out of plane component and bail
+  // if it is farther out of plane than tol.
+  if ( std::abs(p0 * oop) > tol || oop.norm_sq() < tol * tol)
```

Comment:
@dschwen, @jwpeterson, I think this line is triggering a failure for me in small triangles.

I have a pretty small triangle, but it's nearly equilateral:
```c++
 p=(x,y,z)=(   0.001,    0.005,        0)
 e=571
 node 0,   Node id()=275, processor_id()=0, Point=(x,y,z)=(0.000808805, 0.0047284,        0)
 node 1,   Node id()=376, processor_id()=0, Point=(x,y,z)=(0.0011453, 0.00472831,        0)
 node 2,   Node id()=371, processor_id()=0, Point=(x,y,z)=(0.000982249, 0.00508037,        0)
```
I'm pretty sure the test fails simply because v1, v2 are really small, oop is really small.  This isn't a sign of a bad triangle though.  What case are you trying to protect for? 

I think maybe we might need to define a local coordinate system with p0, p1, p2 scaled by the maxiumum bounding box dimension or something to put these in the x,y in [0,1] space or something.

Note a false failure can be triggered by adding

```c++
    a /= 5000.;
    b /= 5000.;
    c /= 5000.;
```

to the unit test tests/mesh/contains_point.C.  Under these conditions the element will fail to find its centroid.

Thoughts?